### ISSUE TITLE
False positive @var is immediately overridden on array

### ISSUE THREAD

**Issue by azhidkov** [kalessil/phpinspectionsea#384]
When working with Elastic Search it's frequent for us adding such items to array:

```php
$params['body']['query']['bool']['must']['filter'][]['range'] = [
    'selected_attributes' => ['gt' => 0]
];

$params['body']['query']['bool']['must']['filter'][]['range'] = [
    'term' => ['category' => $filters['category']]
];
```
Here we add two items to `filter` array, but in second case plugin shows warning:
![screenshot](https://user-images.githubusercontent.com/3435397/27950532-5a1fedba-630a-11e7-8212-e71b65682be7.png)

**Comment by kalessil** [comment]
Thank you for reporting @azhidkov

**Comment by kalessil** [comment]
Fixed
